Sources/AsyncHTTPClient/HTTPClientProxyHandler.swift

```diff
@@ -27,20 +28,28 @@ public extension HTTPClient.Configuration {
     /// TLS will be established _after_ successful proxy, between your client
     /// and the destination server.
     struct Proxy {
+        
+        enum ProxyType: Hashable {
+            case http
+            case socks
+        }
+        
         /// Specifies Proxy server host.
         public var host: String
         /// Specifies Proxy server port.
         public var port: Int
         /// Specifies Proxy server authorization.
         public var authorization: HTTPClient.Authorization?
```

Comment:
I think authorisation should be a property of the `ProxyType`. We have different auth requirements for HTTP and SOCKS proxies.

Reply:
Again I can't really change this without breaking the public API :(

Reply:
Yes you can, but sorry, I wasn't clear.

I mean that `ProxyType` should only store the authorisation when the case is `http`. Then this property can become computed, and crash if it is set when the proxy type is set to `socks`.

Reply:
Understood, thanks!